srcs/executor.c

```diff
@@ -1,24 +1,35 @@
 #include "minishell.h"
 
+/*
+** Case to check later (no command) : 
+	> hello | ls -l
+See if it still segfaults when proper executor is built! 
+*/
+
 int	execute(t_minishell *minishell)
 {
 	int	i;
 
 	i = 0;
 	while (i < minishell->nb_cmds)
 	{
-		if (ft_strncmp(minishell->cmd_table[i].cmd_name, "cd", 2) == 0)
-			minishell->exit_code = cd(&minishell->cmd_table[i], minishell);
-		else if (ft_strncmp(minishell->cmd_table[i].cmd_name, "pwd", 3) == 0)
-			minishell->exit_code = pwd(&minishell->cmd_table[i]);
-		else if (ft_strncmp(minishell->cmd_table[i].cmd_name, "env", 3) == 0)
-			minishell->exit_code = env(&minishell->cmd_table[i], minishell);
-		else if (ft_strncmp(minishell->cmd_table[i].cmd_name, "echo", 4) == 0)
-			minishell->exit_code = echo(&minishell->cmd_table[i]);
-		else if (ft_strncmp(minishell->cmd_table[i].cmd_name, "export", 6) == 0)
-			minishell->exit_code = export(&minishell->cmd_table[i], minishell);
-		// TODO build an actually executor
-		i++;
+		if (!minishell->cmd_table[i].cmd_name)
+			i++; // if cmd_variable is NULL is creates a problem. Not sure if we should always allocate an empty string to CMD anyways if it's empty or handle it here. History should get the line not the cmd
```

Comment:
History is going to be handled right at the beginning I believe, along with readline, from what I understood. So I don't think it's an issue. Don't agree with allocating an empty string, make it NULL (which is what it is now right?) it will be easier to deal with it here

Reply:
I'm a bit confused. You mean we should allocate an empty string if there's no command ? (instead of NULL which it is now)